source/puppet/4.10/lang_data_number.md

```diff
@@ -57,49 +56,63 @@ $product = 8 * 3e5  # evaluates to 240000.0
 
 Integer values can be expressed in decimal notation (base 10), octal notation (base 8), and hexadecimal notation (base 16).
 
-* _Non-zero_ decimal integers must not start with a `0`.
-* Octal values have a prefix of `0`, which can be followed by a sequence of octal digits 0-7.
-* Hexadecimal values have a prefix of `0x` or `0X`, which can be followed by hexadecimal digits 0-9, a-f, or A-F.
+-   _Non-zero_ decimal integers must not start with a `0`.
+-   Octal values have a prefix of `0`, which can be followed by a sequence of octal digits 0-7.
+-   Hexadecimal values have a prefix of `0x` or `0X`, which can be followed by hexadecimal digits 0-9, a-f, or A-F.
 
 Floats can't be expressed in octal or hex.
 
 ``` puppet
-# octal
-$value = 0777   # evaluates to decimal 511
-$value = 0789   # Error, invalid octal
-$value = 0777.3 # Error, invalid octal
-
-# hexadecimal
-$value = 0x777 # evaluates to decimal 1911
-$value = 0xdef # evaluates to decimal 3567
-$value = 0Xdef # same as above
-$value = 0xDEF # same as above
-$value = 0xLSD # Error, invalid hex
+# Octal
+$value = 0777    # evaluates to decimal 511
+$value = 0789    # Error, invalid octal
+$value = 0777.3  # Error, invalid octal
+
+# Hexadecimal
+$value = 0x777  # evaluates to decimal 1911
+$value = 0xdef  # evaluates to decimal 3567
+$value = 0Xdef  # same as above
+$value = 0xDEF  # same as above
+$value = 0xLSD  # Error, invalid hex
 ```
 
 ## Converting numbers to strings
 
 Numbers are automatically converted to strings when interpolated into a string. The automatic conversion uses decimal (base 10) notation.
 
-If you need to convert numbers to non-decimal string representations, you can use [the `sprintf` function.](./function.html#sprintf)
+To convert numbers to non-decimal string representations, you can use the [`sprintf` function](./function.html#sprintf), or you can create a new string object with the `new()` function and [specify its format](./function.html#conversion-to-string).
 
 ## Converting strings to numbers
 
-The [arithmetic operators][arithmetic] will automatically convert strings to numbers.
+[Arithmetic operators][arithmetic] automatically convert strings to numbers.
 
 In all other contexts (resource attributes, function arguments, etc.), Puppet _won't_ automatically convert strings to numbers, but you can:
 
-* Add 0 to manually convert a string to a number. (For example, `$mystring = "85"; $mynum = 0 + $mystring`.)
-* Use [the `scanf` function](./function.html#scanf) to manually extract numbers from strings. This function can also account for surrounding non-numerical text.
+-   Create a new `Integer` or `Float` object using the string as a parameter, such as `$mynum = Integer($mystring)`. This invokes the type's `new()` function, which can convert objects of other types --- including Strings --- into [Integers](./function.html#conversion-to-integer) and [Floats](./function.html#conversion-to-float).
```

Comment:
Perhaps explicitly mention that `Integer.new(args)` is exactly the same as `Integer(args)` or more generically that `T(args)` where `T` is a type name is always interpreted as `T.new(args)`?